lib/nexthop.c

```diff
@@ -36,6 +36,63 @@
 DEFINE_MTYPE_STATIC(LIB, NEXTHOP, "Nexthop")
 DEFINE_MTYPE_STATIC(LIB, NH_LABEL, "Nexthop label")
 
+int nexthop_cmp(const struct nexthop *next1, const struct nexthop *next2)
```

Comment:
Great to have this - but: once it's ready, could we also switch `nexthop_same()` to use it, so we have a single place to maintain the comparison rules?

Reply:
+1 to this.

(note: the new datastructure APIs also get rid of the need for a separate, "inverted" comparison function for hash tables; hash tables just use the same -1/0/+1 there)